src/chocolatey/infrastructure.app/services/TemplateService.cs

```diff
@@ -188,5 +196,162 @@ public void generate_file_from_template(ChocolateyConfiguration configuration, T
+        {
+
+            var pkg = packageManager.LocalRepository.FindPackage("{0}.template".format_with(configuration.TemplateCommand.Name));
+            var pkgVersion = (pkg == null) ? "0.0.0" : pkg.Version.to_string();
+            var pkgTitle = (pkg == null) ? "{0} (Unmanaged)".format_with(configuration.TemplateCommand.Name) : pkg.Title;
+            var pkgSummary = (pkg == null) ? 
+                string.Empty : (pkg.Summary != null && !string.IsNullOrWhiteSpace(pkg.Summary.to_string()) ? "{0}".format_with(pkg.Summary.escape_curly_braces().to_string()) : string.Empty);
+            var pkgDescription = (pkg == null) ? string.Empty : pkg.Description.escape_curly_braces().Replace("\n    ", "\n").Replace("\n", "\n  ");
+            var pkgFiles = "  {0}".format_with(string.Join("{0}  "
+                .format_with(Environment.NewLine), _fileSystem.get_files(_fileSystem
+                    .combine_paths(ApplicationParameters.TemplatesLocation, configuration.TemplateCommand.Name), "*", SearchOption.AllDirectories)));
+
+            var isOverridingBuiltIn = configuration.TemplateCommand.Name == _builtInTemplateOverrideName;
+            var isDefault = string.IsNullOrWhiteSpace(configuration.DefaultTemplateName) ? isOverridingBuiltIn : (configuration.DefaultTemplateName == configuration.TemplateCommand.Name);
+            
+            if (configuration.RegularOutput)
+            {
+                if (configuration.Verbose)
+                {
+                    // TODO list parameters, do later (save to file?)
+                    this.Log().Info(@"Template name: {0}
+Version: {1}
+Default template: {2}
+{3}Title: {4}
+{5}{6}
+List of files:
+{7}
+List of Parameters: 
+  Will be implemented at a later time
+".format_with(configuration.TemplateCommand.Name, 
+                        pkgVersion,
+                        isDefault,
+                        isOverridingBuiltIn ? "This template is overriding the built in template{0}".format_with(Environment.NewLine) : string.Empty, 
+                        pkgTitle,
+                        string.IsNullOrEmpty(pkgSummary) ? "Template not installed as a package" : "Summary: {0}".format_with(pkgSummary), 
+                        string.IsNullOrEmpty(pkgDescription) ? string.Empty : "{0}Description:{0}  {1}".format_with(Environment.NewLine, pkgDescription), 
+                        pkgFiles));
+                }
+                else
+                {
+                    this.Log().Info("{0} {1} {2}".format_with((isDefault ? '*' : ' '), configuration.TemplateCommand.Name, pkgVersion));
+                }
+            }
+            else
+            {
+                this.Log().Info("{0}|{1}".format_with(configuration.TemplateCommand.Name, pkgVersion));
+            }
+        }
+
+        protected void list_built_in_template_info(ChocolateyConfiguration configuration, bool isOverridden, bool isDefault)

```

Comment:
Given that we have decided that there will only ever be one built-in template, do we still _need_ this?

Reply:
No, not necessarily needed. 

I thought it makes for the cleanest implementation, as this code is called from multiple places, and there probably would be duplication if I refactored it.

Reply:
Makes sense!